src/library/scala/concurrent/BatchingExecutor.scala

```diff
@@ -255,6 +255,12 @@ private[concurrent] trait BatchingExecutor extends Executor {
       if (i < BatchingExecutorStatics.syncPreBatchDepth) {
         tl.set(java.lang.Integer.valueOf(i + 1))
         try submitForExecution(runnable) // User code so needs to be try-finally guarded here
+        catch {
+          case ie: InterruptedException =>
+            reportFailure(ie) // TODO: Handle InterruptedException differently?
```

Comment:
If you catch an `InterruptedException` shouldn't you interrupt the current thread again?

Reply:
@WellingR I've been back-and-forth on that one, since control is handed back to the "pool" already. Question is when it should be reset—since we cannot *not execute* any of the `Runnable`s since then they will never be executed.

Reply:
@WellingR I just also remembered: If there's an InterruptedException thrown by user logic, they're using this EC to run blocking logic—which is documented to not be done. :/